### PR TITLE
CommonConfig clean up

### PR DESCRIPTION
**Pull request by ruromero**
* Move auth, mavenRepo and Product out of commonConfig
* Remove the same attributes from CRD
* Rename DB examples
* Added new object `templateConstants` to limit the use of fixed literals

### CODE REVIEW

File: pkg/apis/app/v1/kieapp_types.go [hunk 1]
```diff
@@ -359,17 +366,16 @@ type BuildTemplate struct {
...
-	MavenPassword      string       `json:"mavenPassword,omitempty"`
+	ApplicationName    string `json:"applicationName,omitempty"`
+	Version            string `json:"version,omitempty"`
+	ImageTag           string `json:"imageTag,omitempty"`
+	Product            string `json:"product,omitempty"`
+	KeyStorePassword   string `json:"keyStorePassword,omitempty"`
+	AdminPassword      string `json:"adminPassword,omitempty"`
+	ControllerPassword string `json:"controllerPassword,omitempty"`
+	ServerPassword     string `json:"serverPassword,omitempty"`
+	MavenRepo          string `json:"mavenRepo,omitempty"`
```
Comment (tchughesiv) [inline pkg/apis/app/v1/kieapp_types.go]: should this be moved to template constants? elsewhere?

Comment (ruromero) [reply pkg/apis/app/v1/kieapp_types.go]: I think we should leave this credentials configurable

File: pkg/controller/kieapp/kieapp_controller.go [hunk 2]
```diff
@@ -351,6 +353,29 @@ func (reconciler *Reconciler) dcUpdateCheck(current, new oappsv1.DeploymentConfi
...
+func (reconciler *Reconciler) hasUpdatedVolumes(current, new corev1.PodSpec) bool {
+	cContainer := current.Containers[0]
+	nContainer := new.Containers[0]
+
+	if len(current.Volumes) != len(new.Volumes) || len(cContainer.VolumeMounts) != len(nContainer.VolumeMounts) {
+		return true
+	}
+
+	for i := range current.Volumes {
+		if !strings.HasSuffix(current.Volumes[i].Name, constants.KeystoreVolumeSuffix) && !reflect.DeepEqual(current.Volumes[i], new.Volumes[i]) {
```
Comment (tchughesiv) [inline pkg/controller/kieapp/kieapp_controller.go]: can we expect that the ordering of old and new volume/mount arrays will be identical? should we sort them by name first to ensure proper comparison? i'm not aware of whether ocp does any ordering post deployment.

File: pkg/apis/app/v1/kieapp_types.go (hunk 1 above)
Comment (tchughesiv) [reply pkg/apis/app/v1/kieapp_types.go]: agree, what about  `MavenRepo` though?

File: pkg/controller/kieapp/kieapp_controller.go (hunk 2 above)
Comment (ruromero) [reply pkg/controller/kieapp/kieapp_controller.go]: Removed this logic

**Pull request merged**